craft_application/remote/git.py

```diff
@@ -22,7 +22,24 @@
 from enum import Enum
 from pathlib import Path
 
-import pygit2  # type: ignore[import-untyped]
+# Cannot catch the pygit2 error here raised by the global use of
+# pygit2.Settings on import. We would ideally use pygit2.Settings
+# for this
+try:
+    import pygit2  # type: ignore[import-untyped]
+except Exception:
```

Comment:
I'm accepting suggestions/ideas on how/if we should test this. It's kind of annoying to test import-related errors because this could get imported by a different test in a CI run

Reply:
If it's inconvient or unreliable, I am comfortable not testing this.

One possibility would be to call [`importlib.reload()`](https://docs.python.org/3/library/importlib.html#importlib.reload).

Another possibility is to move the exception handling code to a private function and test the private function.